staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -104,6 +104,12 @@ func SerializeObject(mediaType string, encoder runtime.Encoder, w http.ResponseW
 func WriteObjectNegotiated(ctx request.Context, s runtime.NegotiatedSerializer, gv schema.GroupVersion, w http.ResponseWriter, req *http.Request, statusCode int, object runtime.Object) {
 	serializer, err := negotiation.NegotiateOutputSerializer(req, s)
 	if err != nil {
+		// if original statusCode was not successful we need to return the original error
```

Comment:
@smarterclayton when content negotiation fails, the original errors were getting hidden.  This won't be the content type the receiver wanted, but we're going to be sending them json either way, let's send them the json for the error that failed their request.

Reply:
That's how the spec tells us to do it.  Why should users be allowed to get a different error code here?